client/my-sites/patterns/controller.tsx

```diff
@@ -1,20 +1,46 @@
+import wpcom from 'calypso/lib/wp';
 import Patterns from 'calypso/my-sites/patterns/patterns';
 import { getCurrentUserLocale } from 'calypso/state/current-user/selectors';
 import type { Context as PageJSContext } from '@automattic/calypso-router';
+import type { Pattern } from 'calypso/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/types';
 
 type Next = ( error?: Error ) => void;
 
 export function fetchPatterns( context: PageJSContext, next: Next ) {
-	const locale = getCurrentUserLocale( context.store.getState() ) || context.lang || 'en';
+	const { queryClient, lang, params, store, cachedMarkup } = context;
 
-	// eslint-disable-next-line no-console
-	console.log( '############', locale );
+	if ( cachedMarkup ) {
+		next();
 
-	next();
+		return;
+	}
+
+	const locale = getCurrentUserLocale( store.getState() ) || lang || 'en';
+
+	queryClient
+		.fetchQuery( {
+			queryKey: [ 'patterns', 'library', locale ],
+			queryFn: () => {
+				return wpcom.req.get( `/ptk/patterns/${ locale }`, {
+					categories: 'intro',
+					per_page: '4',
```

Comment:
I've limited the number of results on purpose, so it's faster during development. You can remove those two lines to check how fast it would be to retrieve all the patterns (with my current connection it seems pretty decent).